File: backend/enums.py
from enum import Enum

from tagfiles import ArtistRoles as ArtistRole  # noqa


class ReleaseType(Enum):
    ALBUM = 1
    SINGLE = 2
    EP = 3
    COMPILATION = 4
    SOUNDTRACK = 5
    SPOKENWORD = 6
    LIVE = 7
    REMIX = 8
    DJMIX = 9
    MIXTAPE = 10
    OTHER = 11
    UNKNOWN = 12


class CollectionType(Enum):
    SYSTEM = 1
    COLLAGE = 2
    LABEL = 3
    GENRE = 4
    RATING = 5


class ReleaseSort(Enum):
    RECENTLY_ADDED = "rls.added_on"
    TITLE = "rls.title"
    YEAR = "rls.release_year"
    RANDOM = "RANDOM()"

File: frontend/src/types/enums.ts
export enum ArtistRole {
  MAIN = 'MAIN',
  FEATURE = 'FEATURE',
  REMIXER = 'REMIXER',
  PRODUCER = 'PRODUCER',
  COMPOSER = 'COMPOSER',
  CONDUCTOR = 'CONDUCTOR',
  DJMIXER = 'DJMIXER',
}

export enum ReleaseType {
  ALBUM = 'ALBUM',
  SINGLE = 'SINGLE',
  EP = 'EP',
  COMPILATION = 'COMPILATION',
  SOUNDTRACK = 'SOUNDTRACK',
  SPOKENWORD = 'SPOKENWORD',
  LIVE = 'LIVE',
  REMIX = 'REMIX',
  DJMIX = 'DJMIX',
  MIXTAPE = 'MIXTAPE',
  OTHER = 'OTHER',
  UNKNOWN = 'UNKNOWN',
}

export enum CollectionType {
  SYSTEM = 'SYSTEM',
  COLLAGE = 'COLLAGE',
  LABEL = 'LABEL',
  GENRE = 'GENRE',
  RATING = 'RATING',
}

export enum ReleaseSort {
  RECENTLY_ADDED = 'RECENTLY_ADDED',
  TITLE = 'TITLE',
  YEAR = 'YEAR',
  RANDOM = 'RANDOM',
}

export enum ReleaseView {
  ARTWORK = 'ARTWORK',
  ROW = 'ROW',
}

export type ArtistRoleT = keyof typeof ArtistRole;
export type ReleaseTypeT = keyof typeof ReleaseType;
export type CollectionTypeT = keyof typeof CollectionType;
export type ReleaseSortT = keyof typeof ReleaseSort;
